src/stdio.jl

```diff
@@ -4,13 +4,34 @@
 
 # logging in verbose mode goes to original stdio streams.  Use macros
 # so that we do not even evaluate the arguments in no-verbose modes
+
+#time() returns utc, now() returns local time
+#FYI this could be wrong by -1 second if the now() call happens just before a second boundary and the time() just after
+const utcoffset = floor(Dates.datetime2unix(now())) - floor(time())
+
+function get_log_ts(precision=6)
+    t = time()
+    ms = round(t - floor(t), precision)
+    dt = Dates.unix2datetime(floor(t+utcoffset))
+    msstr = rpad(string(ms)[3:end], precision, "0")
+    ts = "$(string(dt)[12:end]).$msstr"
+end
+
+function get_log_preface()
+    ts = get_log_ts()
+    taskname = get(task_local_storage(), :IJulia_task, "")
+    "$ts($taskname): "
+end
```

Comment:
In the short run, I'd rather just use the simpler code using `now()`; I think for most purposes we won't need logging with sub-second accuracy.

In the long run, it would be good to submit a PR to Base Julia so that `now()` (or some variant thereof) returns milliseconds.  People shouldn't have to resort to ugly code to get a millisecond timestamp in local time.